routers/home.go

```diff
@@ -110,35 +109,17 @@ func RenderRepoSearch(ctx *context.Context, opts *RepoSearchOptions) {
 	}
 
 	keyword := strings.Trim(ctx.Query("q"), " ")
-	if len(keyword) == 0 {
-		repos, count, err = opts.Ranger(&models.SearchRepoOptions{
-			Page:        page,
-			PageSize:    opts.PageSize,
-			Searcher:    ctx.User,
-			OrderBy:     orderBy,
-			Private:     opts.Private,
-			Collaborate: true,
-		})
-		if err != nil {
-			ctx.Handle(500, "opts.Ranger", err)
-			return
-		}
-	} else {
-		if isKeywordValid(keyword) {
-			repos, count, err = models.SearchRepositoryByName(&models.SearchRepoOptions{
-				Keyword:     keyword,
-				OrderBy:     orderBy,
-				Private:     opts.Private,
-				Page:        page,
-				PageSize:    opts.PageSize,
-				Searcher:    ctx.User,
-				Collaborate: true,
-			})
-			if err != nil {
-				ctx.Handle(500, "SearchRepositoryByName", err)
-				return
-			}
-		}
+
+	repos, count, err = models.SearchRepositoryByName(&models.SearchRepoOptions{
+		Page:     page,
+		PageSize: opts.PageSize,
+		OrderBy:  orderBy,
+		Private:  opts.Private,
+		Keyword:  keyword,
```

Comment:
Should we add `Searcher: opts.Searcher,`?

Reply:
`Searcher` is useless without `OwnerID`. In `models.SearchRepositoryByName` is used only if `OwnerID > 0`.